src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -136,22 +136,22 @@ CompilationStartAnalysisContext context
 				SyntaxKind.LocalFunctionStatement
 			);
 
-			context.RegisterSyntaxNodeAction(
+			context.RegisterSymbolAction(
 				ctx => AnalyzeConflictingImmutabilityOnTypeParameters(
 					ctx,
+					(INamedTypeSymbol)ctx.Symbol,
 					annotationsContext
 				),
-				SyntaxKind.TypeParameter
+				SymbolKind.NamedType
 			);

```

Comment:
This was checking all type parameters before, whereas now it will only check type parameters on types.

We already have another check saying `[OnlyIf]` is not allowed a method type parameters, so we don't need to check method type parameters for `[Immutable]` + `[OnlyIf]`, since we don't care about that.